tests/test_transforms/test_decomposition/test_dft_based.py

```diff
@@ -0,0 +1,344 @@
+    with pytest.raises(ValueError, match=f"Feature `target` contains missing values"):
+        transform._check_segments(df=df)
+
+
+def test_check_segments_large_k(ts_with_exogs):
+    df = ts_with_exogs[..., "target"]
+    transform = FourierDecomposeTransform(k=100)
+    with pytest.raises(ValueError, match=f"Parameter `k` must not be greater then"):
+        transform._check_segments(df=df)
+
+
+def test_check_segments_ok(ts_with_exogs):
+    df = ts_with_exogs[..., "target"]
+    transform = FourierDecomposeTransform(k=5)
+    transform._check_segments(df=df)
+
+
+@pytest.mark.parametrize(
+    "series",
+    (
+        pd.Series(np.arange(5)),
+        pd.Series(np.arange(10)),
+        pd.Series([np.nan] * 2 + list(range(5)) + [np.nan] * 3),
+    ),
+)
+def test_fft_components_out_forma(series):
+    expected_columns = ["dft_0", "dft_1", "dft_2", "dft_residuals"]
+    transform = FourierDecomposeTransform(k=3, residuals=True)
+
+    decompose_df = transform._dft_components(series=series)
+
+    assert isinstance(decompose_df, pd.DataFrame)
+    pd.testing.assert_index_equal(decompose_df.index, series.index)
+    assert (decompose_df.columns == expected_columns).all()
+    np.testing.assert_allclose(np.sum(decompose_df.values, axis=1), series.values)
+
+
+def test_is_not_fitted(simple_tsdf):
+    transform = FourierDecomposeTransform(k=5, in_column="feat")
+    with pytest.raises(ValueError, match="Transform is not fitted!"):
+        transform.transform(ts=simple_tsdf)
+
+
+@pytest.mark.parametrize(
+    "ts_name,in_column",
+    (
+        ("outliers_df_with_two_columns", "target"),
+        ("outliers_df_with_two_columns", "feature"),
+        ("ts_with_exogs", "target"),
+        ("ts_with_exogs", "exog"),
+        ("ts_with_exogs", "holiday"),
+        ("example_tsds_int_timestamp", "target"),
+    ),
+)
+def test_fit(ts_name, in_column, request):
+    ts = request.getfixturevalue(ts_name)
+    transform = FourierDecomposeTransform(k=5, in_column=in_column)
+    transform.fit(ts=ts)
+
+    assert transform._first_timestamp == ts.index.min()
+    assert transform._last_timestamp == ts.index.max()
+
+
+@pytest.mark.parametrize("residuals", (True, False))
+@pytest.mark.parametrize("in_column", ("target", "exog"))
+def test_add_residuals(ts_with_exogs, residuals, in_column):
+    ts = ts_with_exogs
+
+    transform = FourierDecomposeTransform(k=5, in_column=in_column, residuals=residuals)
+    transformed = transform.fit_transform(ts=ts)
+
+    assert (f"{in_column}_dft_residuals" in transformed.features) is residuals
+
+
+def test_timestamp_from_history(ts_with_exogs_train_test):
+    test, train = ts_with_exogs_train_test
+    transform = FourierDecomposeTransform(k=5)
+    transform.fit_transform(train)
+
+    with pytest.raises(ValueError, match="First index of the dataset to be transformed must be larger"):
+        transform.transform(test)
+
+
+def test_timestamp_from_future(ts_with_exogs_train_test):
```

Comment:
I can't see any difference with a previous test, why the results are different?

Reply:
How are they different if we are using the same fixture?

Reply:
Different order
`test, train = ts_with_exogs_train_test` in the previos test
`train, test = ts_with_exogs_train_test` here